src/transformers/models/aria/modular_aria.py

```diff
@@ -0,0 +1,1380 @@
+        moe_num_shared_experts (`int`, *optional*, defaults to 2):
+            The number of shared experts.
+        pad_token_id (`int`, *optional*, defaults to 2):
+            The padding token ID.
+    """
+
+    model_type = "aria_text_model"
+    base_config_key = "text_config"
+
+    def __init__(
+        self,
+        moe_intermediate_size: int = 4096,
+        moe_num_experts: int = 8,
+        moe_topk: int = 2,
+        moe_z_loss_coeff: float = 1e-5,
+        moe_aux_loss_coeff: float = 1e-3,
+        moe_num_shared_experts: int = 2,
+        pad_token_id=2,
+        **kwargs,
+    ):
+        super().__init__(pad_token_id=pad_token_id, **kwargs)
+        self.moe_intermediate_size = moe_intermediate_size
+        self.moe_num_experts = moe_num_experts
+        self.moe_topk = moe_topk
+        self.moe_z_loss_coeff = moe_z_loss_coeff
+        self.moe_aux_loss_coeff = moe_aux_loss_coeff
+        self.moe_num_shared_experts = moe_num_shared_experts
+
+
+class AriaConfig(PretrainedConfig):
+    """
+    Configuration class for Aria model.
+
+    This class handles the configuration for both vision and text components of the Aria model,
+    as well as additional parameters for image token handling and projector mapping.
+
+    Args:
+        vision_config (`AriaVisionConfig` or `dict`, *optional*):
+            Configuration for the vision component.
+        vision_feature_layer (`int`, *optional*, defaults to -1):
+            The index of the layer to select the vision feature.
+        text_config (`AriaTextConfig` or `dict`, *optional*):
+            Configuration for the text component.
+        projector_patch_to_query_dict (`dict`, *optional*):
+            Mapping of patch sizes to query dimensions.
+        ignore_index (`int`, *optional*, defaults to -100):
+            Index to ignore in loss calculation.
+        image_token_index (`int`, *optional*, defaults to 32000):
+            Index used to represent image tokens.
+        initializer_range (`float`, *optional*, defaults to 0.02):
+            The standard deviation of the truncated normal initializer for initializing all weight matrices.
+
+    Attributes:
+        model_type (`str`):
+            Type of the model, set to `"aria"`.
+        is_composition (`bool`):
+            Whether the model is a composition of multiple components.
+        ignore_index (`int`):
+            Index to ignore in loss calculation.
+        image_token_index (`int`):
+            Index used to represent image tokens.
+        projector_patch_to_query_dict (`dict`):
+            Mapping of patch sizes to query dimensions.
+        vision_config (`AriaVisionConfig`):
+            Configuration for the vision component.
+        text_config (`AriaTextConfig`):
+            Configuration for the text component.
+    """
+
+    model_type = "aria"
+    is_composition = False
+    sub_configs = {"text_config": AutoConfig, "vision_config": AutoConfig}
```

Comment:
Ok, I change text config to `AriaTextConfig`. But for the vision config, I have to keep AutoConfig, right ? Since I cannot import `Idefic3VisionConfig` in `configuration_aria.py`

Reply:
good question hehe, first time I see it being dependant on another model's config only. Oke, let's keep auto since it is used only in tests now. If we find another use-case, will figure it out in the future